flink-ml-core/src/main/java/org/apache/flink/ml/util/ReadWriteUtils.java

```diff
@@ -118,14 +119,26 @@ public static void saveMetadata(Stage<?> stage, String path) throws IOException
         saveMetadata(stage, path, new HashMap<>());
     }
 
-    /** Returns a subdirectory of the given path for saving/loading model data. */
-    public static String getDataPath(String path) {
-        return Paths.get(path, "data").toString();
+    /**
+     * Returns a subdirectory of the given path for saving/loading table `tableIndex`.
+     *
+     * @param path The parent directory to save the table.
+     * @param tableIndex The index of the table to save.
+     * @return A subdirectory of the given path for saving and loading table `tableIndex`.
+     */
+    public static String getDataPath(String path, int tableIndex) {
```

Comment:
Its better to make this method `private` if it is only used in this class. This is similar to `private static String getPathForPipelineStage(...)`.